src/producer/mock_producer.rs

```diff
@@ -0,0 +1,34 @@
+use rust_arroyo::backends::kafka::config::KafkaConfig;
+use rust_arroyo::backends::kafka::producer::KafkaProducer;
+use rust_arroyo::types::{Topic, TopicOrPartition};
+use sentry_kafka_schemas::Schema;
+use crate::producer::{ExtractCodeError, ResultsProducer};
+use crate::types::result::CheckResult;
+
+pub struct MockResultsProducer {
+    producer: KafkaProducer,
+    topic: TopicOrPartition,
+    schema: Schema,
+}
+
+impl MockResultsProducer {
+    pub fn new(topic_name: &str, config: KafkaConfig) -> Self {
+        let producer = KafkaProducer::new(config);
+        let topic = TopicOrPartition::Topic(Topic::new(topic_name));
+        let schema = sentry_kafka_schemas::get_schema("uptime-results", None).unwrap();
+
```

Comment:
Instead of this actually using kafka for testing, hy don't we have it use a channel, and we can just make sure something got put in the channel in the tests

Reply:
Oh this doesn't use anything for testing, it just uses the kafka stuff to get the schema and validate